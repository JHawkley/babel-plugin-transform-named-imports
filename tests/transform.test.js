const path = require('path');
const pluginTester = require('babel-plugin-tester');
const createBabylonOptions = require('babylon-options');

const plugin = require('../src/index.js');

const babelOptions = {
    filename: 'currentFile.js',
    parserOpts: createBabylonOptions({
        stage: 2
    })
};

pluginTester({
    plugin: () => plugin,
    babelOptions: babelOptions,
    pluginOptions: {
        webpackConfig: path.resolve(__dirname + '/webpack.config.js'),
    },
    snapshot: true,
    tests: {
        // convert this into a default import that leads to `testmodule/myFunc`
        'single named import':
            `import { myFunc } from 'testmodule'`,

        // convert this into named import that leads to `testmodule/myOtherFunc`
        'nested named import':
            `import { myOtherFunc } from 'testmodule'`,

        // convert this into one default import that leads to `testmodule/myFunc`
        // and a named import that leads to `testmodule/myOtherFunc`
        'multiple named imports':
            `import { myFunc, myOtherFunc } from 'testmodule'`,

        // convert this into three imports, one default import for `init` and
        // one default import for `myFunc` and a named one for `myOtherFunc`
        'default import with multiple named import':
            `import init, { myFunc, myOtherFunc } from 'testmodule'`,

        // convert this into a default import with `myAliasFunc` leading to
        // `testmodule/myFunc`
        // BROKEN
        'aliased named import':
            `import { myFunc as myAliasFunc } from 'testmodule'`,

        // don't change existing default imports like this
        'default import':
            `import myFunc from 'testmodule/myFunc'`,
    },
});
