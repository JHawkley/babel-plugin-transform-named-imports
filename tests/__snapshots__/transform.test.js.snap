// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aliased named import 1`] = `
"
import { myFunc as myAliasFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import { myFunc as myAliasFunc } from 'testmodule';
"
`;

exports[`default import 1`] = `
"
import myFunc from 'testmodule/myFunc'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from 'testmodule/myFunc';
"
`;

exports[`default import with multiple named import 1`] = `
"
import init, { myFunc, myOtherFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from './tests/testmodule/myFunc.js';
import { myOtherFunc } from './tests/testmodule/myOtherFunc.js';
"
`;

exports[`multiple named imports 1`] = `
"
import { myFunc, myOtherFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from './tests/testmodule/myFunc.js';
import { myOtherFunc } from './tests/testmodule/myOtherFunc.js';
"
`;

exports[`nested named import 1`] = `
"
import { myOtherFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import { myOtherFunc } from './tests/testmodule/myOtherFunc.js';
"
`;

exports[`single named import 1`] = `
"
import { myFunc } from 'testmodule'

      ↓ ↓ ↓ ↓ ↓ ↓

import myFunc from './tests/testmodule/myFunc.js';
"
`;
